src/common/runShellTasks.ts

```diff
@@ -0,0 +1,87 @@
+/* eslint-disable */
+import child_process from 'child_process';
+import { shellResults } from '../renderer/App.d'
+
+const runDockerComposeKill = (filePath: string) => 
+  runShell(`docker-compose -f ${filePath} kill`, false);
+
+const runDockerComposeListContainer = (filePath: string) => 
+  runShell(`docker-compose -f ${filePath} ps`, false);
+
+const runDockerComposeDeployment = (filePath: string) => 
+  runShell(`docker-compose -f ${filePath} up -d`, false);
+
+const runDockerComposeValidation = (filePath: string) => 
+  runShell(`docker-compose -f ${filePath} config`, true);
+
+const runDockerSwarmInit = (filePath: string) =>
+  runShell(`docker swarm init`, false);
+
+const runDockerSwarmDeployStack = (filePath: string, stackName: string) =>
+  runShell(`docker stack deploy -c ${filePath} ${stackName}`, false);
+
+const runLeaveSwarm = () => 
+  runShell(`docker swarm leave -f`, false);
+
+const runDockerSwarmDeployment = async (filePath: string, stackName: string) => {
+  let stackDeployResult, initResult;
+  await runDockerSwarmInit(filePath).then((data) => initResult = data)
+          .then(() => runDockerSwarmDeployStack(filePath, stackName))
+          .then((info) => {
+            stackDeployResult = info;
+          });
+
+  return JSON.stringify({init: initResult, stackDeploy: stackDeployResult});
+}
+
+const runShell = (cmd: string, filter: boolean) =>
```

Comment:
This shell has not been made generic for all the different commands they're now trying to run (swarm and docker)